src/tiff.imageio/tiffinput.cpp

```diff
@@ -2084,8 +2084,8 @@ TIFFInput::read_native_tile(int subimage, int miplevel, int x, int y, int z,
         return true;
     }
 
-    imagesize_t tile_pixels = m_spec.tile_pixels();
-    imagesize_t nvals       = tile_pixels * m_inputchannels;
+    int tile_pixels = (int)m_spec.tile_pixels();
+    int nvals       = tile_pixels * m_inputchannels;
```

Comment:
These 2 values get passed to `palette_to_rgb` and `bit_convert` which take ints. If int overflows for large tiles, the issue is already there. We may want to modify those two to take a larger type.

This is actually a good example why this work is worth doing.

Reply:
Although each dimension of an image (width, height, depth, nchannels, subimages, etc) are usually declared as int for simplicity (and the fact that at this point, we don't imagine any of those being > 2^31), we use `imagesize_t` (a.k.a. uint64_t) when we have to talk about the number of pixels or values in a tile or images (because those other dimensions may multiply).

So I think in this case, you want to keep tile_pixels and nvals as imagesize_t. It's the count parameters in palette_to_rgb and bit_convert that should probably should be upgraded to imagesize_t rather than int.